barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/ultra_recursive_flavor.hpp

```diff
@@ -345,6 +347,19 @@ template <typename BuilderType> class UltraRecursiveFlavor_ {
                                                    .get_value());
             num_frs_read += num_frs_FF;
 
+            this->contains_recursive_proof = bool(stdlib::field_conversion::convert_from_bn254_frs<CircuitBuilder, FF>(
+                                                      builder, elements.subspan(num_frs_read, num_frs_FF))
+                                                      .get_value());
+            num_frs_read += num_frs_FF;
+
+            for (uint32_t i = 0; i < bb::AGGREGATION_OBJECT_SIZE; ++i) {
```

Comment:
this is the reason I decided to go with the vector->array refactoring. I wanted to be sure that this object was always length 16, and not length 0 or something else, so I could always deserialize it from field elements reliably.